packages/nc-gui/components/workspace/InviteSection.vue

```diff
@@ -7,22 +7,80 @@ const inviteData = reactive({
   roles: WorkspaceUserRoles.VIEWER,
 })
 
+const focusRef = ref<HTMLInputElement>()
+const isDivFocused = ref(false)
+const divRef = ref<HTMLDivElement>()
+
+const emailValidation = reactive({
+  isError: false,
+  message: '',
+})
+
+const validateEmail = (email: string): boolean => {
+  const regEx = /^[^\s@]+@[^\s@]+\.[^\s@]+$/
+  return regEx.test(email)
+}
 const workspaceStore = useWorkspace()
 
 const { inviteCollaborator: _inviteCollaborator } = workspaceStore
 const { isInvitingCollaborators } = storeToRefs(workspaceStore)
 const { workspaceRoles } = useRoles()
 
+// all user input emails are stored here
+const emailBadges = ref<Array<string>>([])
+watch(inviteData, (newVal) => {
+  if (newVal.email.includes(',')) {
+    emailBadges.value.push(newVal.email.split(',')[0])
+    inviteData.email = ''
+  }
+  if (newVal.email.length < 1 && emailValidation.isError) {
```

Comment:
`newVal.email.length < 1` = `!newVal.email.length`. Only one case -> no need to use range.